api/src/plugins/redirect-with-message.test.ts

```diff
@@ -0,0 +1,101 @@
+import Fastify, { FastifyInstance } from 'fastify';
+import qs from 'query-string';
+
+import redirectWithMessage from './redirect-with-message';
+
+async function setupServer() {
+  const fastify = Fastify();
+  await fastify.register(redirectWithMessage);
+  return fastify;
+}
```

Comment:
Can you walk me through the need for us to setupServer, instead of using the one similar to the routes.

https://github.com/ojeytonwilliams/freeCodeCamp/blob/f88b2adebcbdea085b43e3de51a5199075499a82/api/jest.utils.ts#L62-L76

Reply:
Since this is just a plugin, I wanted to test it in isolation. So, this create the bare minimum required for testing: it creates a fastify instance and then it registers just this plugin.

I could have used the full setup, but it seems unnecessary given that we'll be testing any interactions it might have with other plugins when we test the routes that use this.